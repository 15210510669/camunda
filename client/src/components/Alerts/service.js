import {get, post, del, put} from 'request';

export async function loadAlerts() {
  const response = await get('/api/alert');

  return await response.json();
}

export async function loadReports() {
  const response = await get('/api/report');

  return await response.json();
}

export async function saveNewAlert(alert) {
  const response = await post('/api/alert', alert);

  const json = await response.json();

  return json.id;
}

export async function updateAlert(id, alert) {
  return await put('/api/alert/' + id, alert);
}

export async function deleteAlert(id) {
  return await del('api/alert/' + id);
}

