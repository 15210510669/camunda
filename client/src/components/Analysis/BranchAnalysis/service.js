/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import {post} from 'request';

export async function loadFrequencyData(
  processDefinitionKey,
  processDefinitionVersions,
  tenantIds,
  filter
) {
  const response = await post(
    'api/report/evaluate',
    createFlowNodeFrequencyReport(
      processDefinitionKey,
      processDefinitionVersions,
      tenantIds,
      filter
    )
  );

  return await response.json();
}

function createFlowNodeFrequencyReport(
  processDefinitionKey,
  processDefinitionVersions,
  tenantIds,
  filter
) {
  return {
    combined: false,
    reportType: 'process',
    data: {
      processDefinitionKey,
      processDefinitionVersions,
      tenantIds,
      filter,
      view: {
        entity: 'flowNode',
        property: 'frequency',
      },
      groupBy: {
        type: 'flowNodes',
        unit: null,
      },
      visualization: 'heat',
    },
  };
}

export async function loadCorrelationData(
  processDefinitionKey,
  processDefinitionVersions,
  tenantIds,
  filter,
  gateway,
  end
) {
  const response = await post('api/analysis/correlation', {
    processDefinitionKey,
    processDefinitionVersions,
    tenantIds,
    filter,
    gateway,
    end,
  });

  return await response.json();
}
