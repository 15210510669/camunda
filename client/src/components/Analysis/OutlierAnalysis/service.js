/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {get} from 'request';

export async function loadNodesOutliers(config) {
  const response = await get('api/analysis/flowNodeOutliers', config);
  return await response.json();
}

export {loadDurationData, loadCommonOutliersVariables, getInstancesDownloadUrl} from './service.ts';
