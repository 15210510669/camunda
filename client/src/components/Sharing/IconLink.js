/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React, {useEffect, useState} from 'react';

import {getHeader} from 'config';
import {showError} from 'notifications';
import {withErrorHandling} from 'HOC';

import './IconLink.scss';

export function IconLink({mightFail, href}) {
  const [config, setConfig] = useState({});

  useEffect(() => {
    mightFail(getHeader(), setConfig, showError);
  }, [mightFail]);

  return (
    <a className="IconLink" href={href} target="_blank" rel="noopener noreferrer">
      <img src={config.logo} alt="Logo" />
    </a>
  );
}

export default withErrorHandling(IconLink);
