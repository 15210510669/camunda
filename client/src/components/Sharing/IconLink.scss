.IconLink {
  position: fixed;
  height: 20px;
  left: 10px;
  bottom: 10px;
  opacity: 0.5;
  z-index: var(--z-above);

  img {
    width: 20px;
  }
}
