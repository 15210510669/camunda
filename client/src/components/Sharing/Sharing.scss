.Sharing {
  height: 100%;
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  padding: 30px 25px 0 40px;

  .ReportDetails .modalsButtons {
    display: none;
  }

  > .header {
    flex-shrink: 0;
    margin-bottom: 40px;

    .title-button {
      flex-shrink: 0;
    }

    > .InstanceCount {
      margin-top: 0;

      .Popover {
        .countString,
        .filterListHeading {
          display: none;
        }
      }
    }
  }

  .title-container {
    display: flex;
    flex-direction: row;
    flex-grow: 1;
  }

  > .content {
    margin-bottom: 24px;
    display: flex;
    justify-content: flex-start;
    flex-direction: column;
    flex-grow: 1;
    min-height: 0;

    .Table {
      min-width: 25%;
    }
  }

  .OptimizeReport:hover {
    .Popover__button {
      opacity: 1;
    }
  }

  &.compact {
    padding: 10px;

    > .header {
      margin-bottom: 15px;

      .EntityName {
        padding-right: 10px;

        > .name-container .name {
          font-size: 14px;
          font-weight: bold;
          line-height: 30px;
        }

        .Popover {
          margin-top: 4px;
        }
      }

      .title-button {
        border: none;
        box-shadow: none;
        font-size: 14px;
        padding: 5px;
        height: 30px;
        min-width: 55px;

        &:not(:hover):not(.active):not(:active) {
          background: transparent;
        }
      }
    }

    > .content {
      margin-bottom: 0;
    }
  }
}
