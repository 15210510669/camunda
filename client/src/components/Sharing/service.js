/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import {get} from 'request';

export async function evaluateEntity(id, type) {
  let response;

  try {
    response = await get(`api/share/${type}/${id}/evaluate`);
  } catch (e) {
    return null;
  }

  return await response.json();
}

export function createLoadReportCallback(dashboardShareId) {
  return async reportId => {
    try {
      const response = await get(
        `api/share/dashboard/${dashboardShareId}/report/${reportId}/evaluate`
      );
      return await response.json();
    } catch (error) {
      return (await error.json()).reportDefinition;
    }
  };
}
