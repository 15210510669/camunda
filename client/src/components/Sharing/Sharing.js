import React from 'react';

import './Sharing.css';
import {ReportView} from 'components';
import {getReportData} from './service';


export default class Sharing extends React.Component {
  constructor(props) {
    super(props);

    this.id = props.match.params.id;

    this.state = {
      reportResult: null,
      loaded: false
    };

    this.loadReport();
  }

  loadReport = async () => {

    const reportResult = await getReportData(this.id);

    this.setState({
      reportResult,
      loaded: true
    });
  }

  render() {
    const {loaded, reportResult} = this.state;
    if(!loaded) {
      return <div className='Sharing__loading-indicator'>loading...</div>;
    }

    if(!reportResult) {
      return <div className='Sharing__error-message'>The resource you want to access is not available!</div>;
    }

    return (<div style={{flexDirection: 'column', display: 'flex', flexGrow: 1}}>
    <div className='Sharing__header'>
          <div className='Sharing__title-container'>
            <h1 className='Sharing__tilte'>{reportResult.report.name}</h1>
          </div>
        </div>
      <div className='Sharing__content' >
          <ReportView report={reportResult.report} />
     </div>
    </div>);
  }
}