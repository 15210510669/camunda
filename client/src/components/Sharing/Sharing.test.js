/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import {shallow} from 'enzyme';

import {Button, ReportRenderer} from 'components';

import Sharing from './Sharing';
import {evaluateEntity} from './service';

jest.mock('./service', () => {
  return {
    evaluateEntity: jest.fn(),
    createLoadReportCallback: jest.fn(),
  };
});

const props = {
  match: {
    params: {
      id: 123,
    },
  },
};

it('should render without crashing', () => {
  shallow(<Sharing {...props} />);
});

it('should initially load data', () => {
  shallow(<Sharing {...props} />);

  expect(evaluateEntity).toHaveBeenCalled();
});

it('should display a loading indicator', () => {
  const node = shallow(<Sharing {...props} />);

  expect(node.find('LoadingIndicator')).toExist();
});

it('should display an error message if evaluation was unsuccessful', () => {
  props.match.params.type = 'report';
  const node = shallow(<Sharing {...props} />);

  node.setState({
    loading: false,
    evaluationResult: null,
  });

  expect(node.find('ErrorPage')).toExist();
});

it('should display an error message if type is invalid', () => {
  props.match.params.type = 'foo';
  const node = shallow(<Sharing {...props} />);

  node.setState({
    loading: false,
    evaluationResult: {name: 'foo'},
  });

  expect(node.find('ErrorPage')).toExist();
});

it('should have report if everything is fine', () => {
  props.match.params.type = 'report';
  const node = shallow(<Sharing {...props} />);

  node.setState({
    loading: false,
    evaluationResult: {name: 'foo'},
  });

  expect(node.find(ReportRenderer)).toExist();
});

it('should retrieve report for the given id', () => {
  props.match.params.type = 'report';
  shallow(<Sharing {...props} />);

  expect(evaluateEntity).toHaveBeenCalledWith(123, 'report');
});

it('should display the report name', () => {
  props.match.params.type = 'report';
  const node = shallow(<Sharing {...props} />);

  node.setState({
    loading: false,
    evaluationResult: {name: 'My report name'},
  });

  expect(node).toIncludeText('My report name');
});

it('should have dashboard if everything is fine', () => {
  props.match.params.type = 'dashboard';
  const node = shallow(<Sharing {...props} />);

  node.setState({
    loading: false,
    evaluationResult: {reportShares: 'foo'},
  });

  expect(node.find('DashboardRenderer')).toExist();
});

it('should retrieve dashboard for the given id', () => {
  props.match.params.type = 'dashboard';
  shallow(<Sharing {...props} />);

  expect(evaluateEntity).toHaveBeenCalledWith(123, 'dashboard');
});

it('should display the dashboard name', () => {
  props.match.params.type = 'dashboard';
  const node = shallow(<Sharing {...props} />);

  node.setState({
    loading: false,
    evaluationResult: {name: 'My dashboard name'},
  });

  expect(node).toIncludeText('My dashboard name');
});

it('should render a button linking to view mode', () => {
  props.match.params.type = 'dashboard';
  const node = shallow(<Sharing {...props} />);

  node.setState({
    loading: false,
    evaluationResult: {name: 'My dashboard name'},
  });

  expect(node.find(Button)).toExist();
});
