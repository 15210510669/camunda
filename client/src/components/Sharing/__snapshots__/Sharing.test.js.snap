// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show a compact version and lock scroll when a shared report is embedded 1`] = `
<div
  className="Sharing compact"
>
  <div
    className="header"
  >
    <div
      className="title-container"
    >
      <EntityName
        details={
          <ReportDetails
            report={
              Object {
                "id": "aReportId",
                "name": "My report name",
              }
            }
          />
        }
      >
        My report name
      </EntityName>
      <a
        className="Button main title-button"
        href="http://example.com/subUrl/#/report/aReportId/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <Icon
          renderedIn="span"
          type="share"
        />
        <span>
          Open
        </span>
      </a>
    </div>
    <InstanceCountErrorHandler
      report={
        Object {
          "id": "aReportId",
          "name": "My report name",
        }
      }
    />
  </div>
  <div
    className="content"
  >
    <Memo(ReportRenderer)
      context="shared"
      error={null}
      loadReport={[Function]}
      report={
        Object {
          "id": "aReportId",
          "name": "My report name",
        }
      }
    />
    <IconLinkErrorHandler
      href="http://example.com/subUrl/#/report/aReportId/"
    />
    <DiagramScrollLock />
  </div>
</div>
`;
